File: backend/src/main/java/com/musaugurlu/luckylibrary/backend/services/BookService.java
/*
 * Copyright (c) 2020.
 * Author: Musa Ugurlu
 * Date: 7/3/20, 1:55 PM
 */

package com.musaugurlu.luckylibrary.backend.services;

import com.musaugurlu.luckylibrary.backend.models.Book;
import com.musaugurlu.luckylibrary.backend.models.Category;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface BookService {
    Optional<Book> findById(UUID id);
    List<Book> findAll();
    Book save(Book book);
    List<Book> findByTitle(String title);
    Optional<Book> findFirstBy();
    Optional<List<Book>> getNewBooks();
    Optional<List<Book>> getPopularBooks();
    long count();
    List<Book> search(String word);
    List<Book> findAllByCategory(Category category);
}

File: frontend/src/services/bookService.js
/*
 * Copyright (c) 2020 Musa Ugurlu
 * Author: Musa Ugurlu
 * Date: 10/03/2020 10:16:45 pm
 */

import axios from 'axios'

const backend_url = process.env.VUE_APP_API_BASE_URL;

export const bookService = {
    getNewBooks,
}

function getNewBooks() {
    axios.get(`${backend_url}/books/newbooks`)
        .then(res => {
            return res.data;
        })
}